src/Kestrel.Core/Internal/Http2/Http2Stream.FeatureCollection.cs

```diff
@@ -246,13 +246,13 @@ MinDataRate IHttpMinResponseDataRateFeature.MinDataRate
 
         object IFeatureCollection.this[Type key]
         {
-            get => FastFeatureGet(key);
```

Comment:
I'm not so sure connection-level features should be reset be request. Then again, I'm not sure why you would want to mutate a connection-level feature via the request's feature collection. If there was a way to make these features immutable when accessed this way, we probably would.